### PR TITLE
Transactions pulled from chain

### PR DESCRIPTION
**Pull request by piggydoughnut**
- all parsing for transactions is done on the bitgo side and will be reviewed by dimxy
- txs are pulled from chain instead of the explorer
- updates in the worker file in terms of connection due to bitgo updates

These changes were necessary as a preparation for #138

### CODE REVIEW

**Comment by piggydoughnut** [comment]
@acnebs this code includes those changes with connection. The function I am using is this one https://github.com/TokelPlatform/bitgo-komodo-cc-lib/blob/development/net/connect.js 
You can see the error listeners in there. Those are the ones which will notify us of errors with peers. 

I have created a separate issue for this connection task here #145

File: src/electron/worker.js [hunk 1]
```diff
@@ -1,59 +1,33 @@
...
-        resolve(true);
-      });
-    }).catch(e => console.log(e));
+    if (process.env.NODE_ENV === 'test') {
+      return 'singleton created';
+    }
+    try {
+      this.peers = await nspvConnect({ network }, {});
+      this.connected = true;
+      console.log('asdasdas', this.peers);
```
Comment (acnebs) [inline src/electron/worker.js:22]: Gross.

File: src/electron/worker.js [hunk 2]
```diff
@@ -152,76 +127,64 @@ class NspvBitGoSingleton {
...
-        numutxos: response.utxos.length,
-        address: this.address,
-      };
-    } catch (e) {
-      console.error(e);
-      throw new Error(e);
+    if (!this.connected) {
+      throw new Error('Not connected');
     }
+    const response = await ccutils.getNormalUtxos(this.peers, address, 0, 0);
```
Comment (acnebs) [inline src/electron/worker.js:133]: Should await these in parallel with `Promise.all`.

File: src/electron/worker.js [hunk 3]
```diff
@@ -152,76 +127,64 @@ class NspvBitGoSingleton {
...
-      };
-    } catch (e) {
-      console.error(e);
-      throw new Error(e);
+    if (!this.connected) {
+      throw new Error('Not connected');
     }
+    const ccUtxos = await ccutils.getCCUtxos(this.peers, address);
+    return {
+      ...ccUtxos,
```
Comment (acnebs) [inline src/electron/worker.js:152]: Why spread `ccUtxos` instead of just returning it directly?

File: src/util/transactions.ts [hunk 4]
```diff
@@ -46,6 +47,21 @@ export const parseListTxsRpcTx = tx => {
...
 };
 
+export const parseBlockchainTransaction = (tx, address: string) => {
+  const iAmSender = tx.senders.find(s => s === address);
+  return [
+    {
+      value: toBitcoin(tx.value),
+      from: tx.senders,
+      recipient: tx.recepients,
+      time: moment.unix(tx.time).format('DD/MM/YYYY H:mm:ss'),
```
Comment (acnebs) [inline src/util/transactions.ts:57]: 1. Let's not use `moment`, `date-fns` is much better.
2. Probably better to just leave this as a unix timestamp and only format directly in components where we want to display it, which allows us to more easily display in different ways in different places.

**Review by acnebs — CHANGES_REQUESTED** [review]

File: src/electron/worker.js (hunk 3 above)
Comment (piggydoughnut) [reply src/electron/worker.js:152]: Oh I must have been adding something else in there before. Will change.

File: src/electron/worker.js (hunk 2 above)
Comment (piggydoughnut) [reply src/electron/worker.js:133]: actually need to check if we can use that getCCUtxos function at all

**Pull request merged**